packages/x-components/src/x-modules/facets/components/lists/exclude-filters-with-no-results.vue

```diff
@@ -17,27 +15,51 @@
    *
    * @public
    */
-  @Component({
-    mixins: [xComponentMixin(facetsXModule)]
-  })
-  export default class ExcludeFiltersWithNoResults extends mixins(FiltersInjectionMixin) {
-    /**
-     * Removes the filters that have exactly 0 results associated.
-     *
-     * @returns A sublist of the filters prop, excluding the ones with no results.
-     * @internal
-     */
-    @XProvide('filters')
-    public get filtersWithResults(): Filter[] {
-      return this.renderedFilters.filter(
-        filter => !isBooleanFilter(filter) || filter.totalResults !== 0
-      );
-    }
+  export default defineComponent({
+    name: 'ExcludeFiltersWithNoResults',
+    xModule: facetsXModule.name,
+    props: {
+      /**
+       * The list of filters to be rendered as slots.
+       *
+       * @public
+       */
+      filters: Array as PropType<Filter[]>,
+
+      /**
+       * This prop is used in the `HierarchicalFilter` component and only in that case. It is necessary
+       * to make the `renderedFilters` to return only the filters of each level of the hierarchy.
+       *
+       * @public
+       */
+      parentId: {
+        type: String as PropType<Filter['id']>,
+        required: false
+      }
+    },
+    setup(props, { slots }) {
+      useRegisterXModule(facetsXModule);
+      const renderedFilters = useFiltersInjection(props);
 
+      /**
+       * Removes the filters that have exactly 0 results associated.
+       *
+       * @returns A sublist of the filters prop, excluding the ones with no results.
+       * @internal
+       */
+      const filtersWithResults = computed((): Filter[] => {
+        return renderedFilters.value.filter(
+          filter => !isBooleanFilter(filter) || filter.totalResults !== 0
+        );
+      });
+      provide('filters', filtersWithResults);
+
+      return { filtersWithResults, slots };
+    },
     render(h: CreateElement): VNode {
-      return this.$scopedSlots.default?.({ filters: this.filtersWithResults })?.[0] ?? h();
+      return this.slots.default?.({ filters: this.filtersWithResults })?.[0] ?? h();
```

Comment:
Yeah, pretty much I believe. Since in this case we are not using a template we can just [declare the render function](https://vuejs.org/guide/extras/render-function.html#declaring-render-functions) by returning the render function in the setup. 

`return () => (slots.default ? slots.default({ filters: filtersWithResults.value }) : h());`

I think we don't need to just get the first element but it is safe to return the array of VNodes from the default slot directly